csrc/ir/utils.cpp

```diff
@@ -752,6 +752,15 @@ bool isSelectInput(TensorView* tv) {
   return false;
 }
 
+bool isSliceInput(TensorView* tv) {
+  for (auto expr : tv->uses()) {
+    if (expr->isA<SliceOp>()) {
```

Comment:
I don't think padding inputs would have the same issue (unless doing something crazy like padding with negative widths). I think the problem here happens because in the original fusion a reduction tensor was picked as the reference, so its shape determined the thread block size, but that resulted in an invalid block size for the input. Avoiding caching the input would prevent the error from happening, but the underlying issue is what I initially did for these schedulers to (appear to) work with resize ops. 

Specifically, the assumption we have here is that all the other tensors are either the same shape as the reference or lack some domains due to broadcasting. This is not the case with IterDomain resizing, but initially I just patched the schedulers to ignore the extent differences so that everything just work. In this case, the number of threads was determined based on the post-slice tensor, but that resulted in too large for the pre-resize tensor. 

Padding an input believe should be fine as the input should be smaller than the padding output (unless it's negative padding).

More generally, this problem would happen when a reference tensor is smaller than tensors before slicing or after padding and boils down to the question of what the effective scheduling should look like for fusions with different shapes.

Reply:
Since slice inputs must be a segment input, disabling caching should work around the problem, so I think this fix is a reasonable WAR for now. Thanks @liqiangxl.

Reply:
Thanks for the background info.
Here what I observed is: the launch parameter is based on the maximum dim. 
For example, TIDx is used to parallel
`ithreadIdx.x190{( ceilDiv(( ceilDiv(( ceilDiv(( i246 * i247 ), 4) ), 8) ), 1) )}_p` in the slice tv
 and
`rthreadIdx.x106{( ceilDiv(( ceilDiv(( ceilDiv(4096, 4) ), 8) ), 1) )}_p ` in the reduction tv. 
The value of bdimx = max(4096/4/8, ( i246 * i247 )/4/8).
When `( i246 * i247 )/4/8` is less than 1024, the kernel runs without error, just not efficient. This is why I need to set ` i246 * i247` to a large value to trigger the original error.

Reply:
> something crazy like padding with negative widths

That's what I had in mind since one can effectively do a slice this way, but you're right it's probably rare and could possibly separated/translated in integration, so I agree it's probably fine to ignore `PadOp` here.

Reply:
`PadOp` is added to the skip list.